### PR TITLE
Validator and Blobber Update Settings with new Models

### LINKED ISSUE #2544: Update provider, use map as input rather than object
Update provider smart contract endpoints currently use an object as input. This can cause an issue distinguishing between unchanged and default settings. 

A suggestion; we change the input to the update_provider endpoints to take a map of updated fields as input rather than an object. 

Will need `zbox` ad `zwallet` changes as well.

### PR DESCRIPTION
**Pull request by sunilmhta**
## Fixes
#2544 

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk: https://github.com/0chain/gosdk/pull/1095
- system_test:
- zboxcli: https://github.com/0chain/zboxcli/pull/480
- zwalletcli: 
- 0chain: https://github.com/0chain/0chain/pull/2596

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zboxcli  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zwalletcli  | sprint-july-1  |
| 0chain  | users/sunilmhta/fix/2544/0chain/blobber-settings  |
| blobber  | sprint-july-1  |
| authorizer  | sprint-july-1  |
| 0box  | sprint-july-1  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zboxcli  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zwalletcli  | sprint-july-1  |
| 0chain  | users/sunilmhta/fix/2544/0chain/blobber-settings  |
| blobber  | sprint-july-1  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zboxcli  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zwalletcli  | staging  |
| 0chain  | users/sunilmhta/fix/2544/0chain/blobber-settings  |
| blobber  | sprint-july-1  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zboxcli  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zwalletcli  | sprint-july-1  |
| 0chain  | users/sunilmhta/fix/2544/0chain/blobber-settings  |
| blobber  | sprint-july-1  |
| authorizer  | sprint-july-1  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zboxcli  | users/sunilmhta/fix/2544/updateblobnvalidato  |
| zwalletcli  | sprint-july-1  |
| 0chain  | users/sunilmhta/fix/2544/0chain/blobber-settings  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | sprint-july-1  |
| gosdk  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zboxcli  | users/sunilmhta/fix/2544/updateblobnvalidator  |
| zwalletcli  | sprint-july-1  |
| 0chain  | users/sunilmhta/fix/2544/0chain/blobber-settings  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

File: code/go/0chain.net/smartcontract/storagesc/blobber.go [hunk 1]
```diff
@@ -72,102 +74,207 @@ func (sc *StorageSmartContract) hasBlobberUrl(blobberURL string,
...
+	updateBlobberRequest *dto.StorageDtoNode,
+	existingBlobber *StorageNode,
+	existingStakePool *stakePool,
 	balances cstate.StateContextI,
 ) (err error) {
-	// check terms
-	if err = blobber.Terms.validate(conf); err != nil {
-		return fmt.Errorf("invalid blobber terms: %v", err)
+	// validate the new terms and update the existing blobber's terms
+	if err = validateTermsInUpdatedRequestAndMergeInSavedBlobber(updateBlobberRequest, existingBlobber, conf); err != nil {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/blobber.go]: we used to have shorter names in golang `validateAndSave` can be good as well

File: code/go/0chain.net/smartcontract/storagesc/blobber.go [hunk 2]
```diff
@@ -72,102 +74,207 @@ func (sc *StorageSmartContract) hasBlobberUrl(blobberURL string,
...
+		return err
 	}
 
-	if blobber.Capacity <= 0 {
-		return sc.removeBlobber(t, blobber, balances)
+	if err = validateGeolocationInUpdatedRequestAndMergeInSavedBlobber(updateBlobberRequest, existingBlobber); err != nil {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/blobber.go]: same here, just a short quote from effective go

```
Another short example is once.Do; once.Do(setup) reads well and would not be improved by writing once.DoOrWaitUntilDone(setup). Long names don't automatically make things more readable. A helpful doc comment can often be more valuable than an extra long name.
```

File: code/go/0chain.net/smartcontract/storagesc/blobber.go [hunk 3]
```diff
@@ -72,102 +74,207 @@ func (sc *StorageSmartContract) hasBlobberUrl(blobberURL string,
...
 	sc.statIncr(statUpdateBlobber)
 
-	if savedBlobber.Capacity == 0 {
+	if currentCapacity == 0 {
 		sc.statIncr(statNumberOfBlobbers) // reborn, if it was "removed"
 	}
 
-	if err = validateStakePoolSettings(blobber.StakePoolSettings, conf); err != nil {
-		return fmt.Errorf("invalid new stake pool settings:  %v", err)
+	if err = validateStakePoolSettingsInUpdatedRequestAndMergeInSavedBlobber(
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/blobber.go]: make it shorter

File: code/go/0chain.net/smartcontract/storagesc/blobber.go [hunk 4]
```diff
@@ -273,29 +381,27 @@ func (sc *StorageSmartContract) updateBlobberSettings(t *transaction.Transaction
 			"can't get the blobber: "+err.Error())
 	}
 
-	var sp *stakePool
-	if sp, err = sc.getStakePool(spenum.Blobber, updatedBlobber.ID, balances); err != nil {
+	var existingStakePool *stakePool
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/blobber.go]: `sp` was good, as we don't have any other stakePool in this func

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/blobber.go (hunk 2 above)
Comment (sunilmhta) [reply code/go/0chain.net/smartcontract/storagesc/blobber.go]: Renaming to validateAndSaveGeoLoc() and the terms func to validateAndSaveTerms()

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  |   |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  | sprint-july-2  |
| blobber  |   |
| authorizer  |   |
| 0box  |   |
| 0dns  |   |

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**